processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -391,4 +392,39 @@ public static SegmentAnalysis finalizeAnalysis(SegmentAnalysis analysis)
         analysis.isRollup()
     );
   }
+
+  public EnumSet<SegmentMetadataQuery.AnalysisType> getAnalysisTypes(SegmentMetadataQuery query)
+  {
+    if(query.getAnalysisTypes().hashCode() == SegmentMetadataQuery.DEFAULT_ANALYSIS_TYPES.hashCode() && config != null) {
+      return config.getDefaultAnalysisType();
+    }
+    return query.getAnalysisTypes();
+  }
+
+  public SegmentAnalyzer getSegmentAnalyzer(SegmentMetadataQuery query)
+  {
+    EnumSet<SegmentMetadataQuery.AnalysisType> analysisTypes = getAnalysisTypes(query);
```

Comment:
may as well make this one line with `return new SegmentAnalyzer(getAnalysisTypes(query));`